UnchainedLauncherCore/src/Installer/UnchainedLauncherInstaller.cs

```diff
@@ -0,0 +1,267 @@
+
+                return ReleaseCache.ToList().Flatten();
+            } catch (Exception e) {
+                logger.Error("Failed to connect to github to retrieve version information", e);
+                return ImmutableList.CreateBuilder<VersionedRelease>().ToImmutableList();
+            }
+        }
+
+        /// <summary>
+        /// Installs the launcher by downloading the latest release and replacing the current executable with the new one.
+        /// </summary>
+        /// <param name="targetDir"></param>
+        /// <param name="release"></param>
+        /// <param name="replaceCurrent">If true, closes the current executable and launches the installed executable with the same args used to launch the current executable.</param>
+        /// <returns>
+        /// </returns>
+        public async Task<bool> Install(DirectoryInfo targetDir, VersionedRelease release, bool replaceCurrent, Action<string>? logProgress = null) {
+            var log = new Action<string>(s => {
+                logProgress?.Invoke(s);
+                logger.Info(s);
+            });
+
+            try {
+                var url =
+                    (from releaseAssets in release.Release.Assets
+                     where releaseAssets.Name.Contains("Launcher.exe")
+                     select releaseAssets.BrowserDownloadUrl).First();
+
+
+                var fileName = $"UnchainedLauncher-{release.Version}.exe";
+                var downloadFilePath = Path.Combine(targetDir.FullName, fileName);
+
+                log($"Downloading release {release.Release.TagName}\n    from {url}\n    to {downloadFilePath}");
+
+                // We only want to download the Launcher executable, even if the release contains multiple assets
+                var assetFilter = (ReleaseAsset asset) => asset.Name.Contains("Launcher.exe") ? Some(downloadFilePath) : None;
+                var downloadResult = await DownloadRelease(release, assetFilter, log);
+
+                if (!downloadResult) {
+                    log($"Failed to download the launcher version {release.Version}.");
+                    return false;
+                }
+                
+                var launcherPath = Path.Combine(targetDir.FullName, FilePaths.LauncherPath);
+
+                if (replaceCurrent) {
+                    var currentExecutableName = Process.GetCurrentProcess().ProcessName;
+
+                    if (!currentExecutableName.EndsWith(".exe")) {
+                        currentExecutableName += ".exe";
+                    }
+
+                    log($"Replacing current executable {currentExecutableName} with downloaded launcher {downloadFilePath}");
+
+                    var commandLinePass = string.Join(" ", Environment.GetCommandLineArgs().Skip(1));
+
+                    var powershellCommand = new List<string>() {
+                        $"Wait-Process -Id {Environment.ProcessId}",
+                        $"Start-Sleep -Milliseconds 1000",
```

Comment:
This is giving time to release resources, right? Maybe replace with a while loop testing for locked file? Otherwise it may fail on actual wooden pcs I think

Reply:
This is how it's been done since the beginning for self-updates. I don't really feel like messing with it because it's already slightly jank

Reply:
I think I tried looking for something like that but the process of checking was annoying and 1 second worked. It'd definitely be better to actually check, though.